src/app/public/modules/flyout/flyout.component.spec.ts

```diff
@@ -869,4 +869,110 @@ describe('Flyout component', () => {
+    }));
+
+    it('should add the sm class when appropriate', fakeAsync(() => {
+      openFlyout({ maxWidth: 10000, minWidth: 50 });
+      fixture.detectChanges();
+      tick();
+      const flyoutElement = getFlyoutElement();
+      const handleElement = getFlyoutHandleElement();
+
+      expect(flyoutElement.style.width).toBe('500px');
+
+      let evt = document.createEvent('MouseEvents');
+      evt.initMouseEvent('mousedown', false, false, window, 0, 0, 0, 1000,
+        0, false, false, false, false, 0, undefined);
+
+      handleElement.dispatchEvent(evt);
+      makeEvent('mousemove', { clientX: 600 });
+      fixture.detectChanges();
+      tick();
+      makeEvent('mouseup', {});
+
+      expect(flyoutElement.style.width).toBe('900px');
+      expect(flyoutElement.classList.contains('sky-responsive-container-sm')).toBeTruthy();
+      expect(flyoutElement.classList.contains('sky-responsive-container-xs')).toBeFalsy();
+      expect(flyoutElement.classList.contains('sky-responsive-container-md')).toBeFalsy();
+      expect(flyoutElement.classList.contains('sky-responsive-container-lg')).toBeFalsy();
+    }));
+
+    it('should add the md class when appropriate', fakeAsync(() => {
+      openFlyout({ maxWidth: 10000, minWidth: 50 });
+      fixture.detectChanges();
+      tick();
+      const flyoutElement = getFlyoutElement();
+      const handleElement = getFlyoutHandleElement();
+
+      expect(flyoutElement.style.width).toBe('500px');
+
+      let evt = document.createEvent('MouseEvents');
+      evt.initMouseEvent('mousedown', false, false, window, 0, 0, 0, 1000,
+        0, false, false, false, false, 0, undefined);
+
+      handleElement.dispatchEvent(evt);
+      makeEvent('mousemove', { clientX: 400 });
+      fixture.detectChanges();
+      tick();
+      makeEvent('mouseup', {});
+
+      expect(flyoutElement.style.width).toBe('1100px');
+      expect(flyoutElement.classList.contains('sky-responsive-container-md')).toBeTruthy();
+      expect(flyoutElement.classList.contains('sky-responsive-container-xs')).toBeFalsy();
+      expect(flyoutElement.classList.contains('sky-responsive-container-sm')).toBeFalsy();
+      expect(flyoutElement.classList.contains('sky-responsive-container-lg')).toBeFalsy();
+    }));
+
+    it('should add the lg class when appropriate', fakeAsync(() => {
+      openFlyout({ maxWidth: 10000, minWidth: 50 });
+      fixture.detectChanges();
+      tick();
+      const flyoutElement = getFlyoutElement();
+      const handleElement = getFlyoutHandleElement();
+
+      expect(flyoutElement.style.width).toBe('500px');
+
+      let evt = document.createEvent('MouseEvents');
+      evt.initMouseEvent('mousedown', false, false, window, 0, 0, 0, 1000,
+        0, false, false, false, false, 0, undefined);
+
+      handleElement.dispatchEvent(evt);
+      makeEvent('mousemove', { clientX: 100 });
+      fixture.detectChanges();
+      tick();
+      makeEvent('mouseup', {});
```

Comment:
I feel like we could modularize the mouse movement here a bit. What would you say about making a single function like `dragFlyoutHandle(dragX: number)`?

Reply:
Done. 

I ended up making three function for 
1. grabbing the handle
2. moving the handle
3. releasing the handle

and then I made a fourth function that will do all three. 

There were a couple cases where we only needed two of the steps (to intentionally test what happens when the other is left out) and so that is why I didn't do everything in just one function.

Reply:
Much nicer! Well done.